src/CommunityToolkit.Maui.SourceGenerators/Generators/TextColorToGenerator.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using CommunityToolkit.Maui.SourceGenerators.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CommunityToolkit.Maui.SourceGenerators.Generators;
+
+[Generator]
+class TextColorToGenerator : IIncrementalGenerator
+{
+	public void Initialize(IncrementalGeneratorInitializationContext context)
+	{
+		// Get All Classes in User Library
+		var candidateClassesProvider = context.SyntaxProvider.CreateSyntaxProvider(
+			(syntaxNode, cancellationToken) => syntaxNode is ClassDeclarationSyntax,
+			(context, cancellationToken) => (ClassDeclarationSyntax)context.Node);
+
+		// Get Microsoft.Maui.Controls Assymbly Symbol
+		var mauiControlsAssemblySymbolProvider = context.CompilationProvider.Select(
+			(compilation, token) => compilation.SourceModule.ReferencedAssemblySymbols.Single(q => q.Name == "Microsoft.Maui.Controls"));
+
+		var inputs = candidateClassesProvider.Collect()
+						.Combine(mauiControlsAssemblySymbolProvider)
+						.Select((combined, cancellationToken) => (CandidateClassesProvider: combined.Left, MauiControlsAssemblySymbolProvider: combined.Right));
+		context.RegisterSourceOutput(inputs, (context, collectedValues) =>
+		{
+			var textStyleClassList = new List<(string ClassName, string Namespace)>();
+
+			// Collect Microsoft.Maui.Controls that Implement ITextStyle
+			foreach (var namedTypeSymbol in collectedValues.MauiControlsAssemblySymbolProvider.GlobalNamespace.GetNamedTypeSymbols())
+			{
+				if (namedTypeSymbol.ImplementsInterfaceOrBaseClass("ITextStyle"))
+				{
+					textStyleClassList.Add((namedTypeSymbol.Name, namedTypeSymbol.ContainingNamespace.ToString()));
+				}
+			}
+			// Collect All Classes in User Library that Implement ITextStyle
+			foreach (var classDeclarationSyntax in collectedValues.CandidateClassesProvider)
+			{
+				if (classDeclarationSyntax.BaseList?.Types.Any(x => x.ToString() == "ITextStyle") is true)
```

Comment:
**Strings are evil**
So, if the user has his own interface `ITextStyle` it will be added to `textStyleClassList`.
In other words, let say that my project or a lib that I consume has an interface called `ITextStyle`, as you can see in the image below

![image](https://user-images.githubusercontent.com/20712372/147892427-5a2067ad-8f7e-4130-a032-ad27d68cf101.png)


Them when we iterate on our SG we will have that class to generate, as you can see in the image below
![image](https://user-images.githubusercontent.com/20712372/147892093-39969ee8-97c6-425d-9f36-e21e22db6e98.png)

And then our SG will fail.

I would suggest you work with `Symbols` instead of `strings`, the solution will be something like:

```csharp
 var textStyleClassList = new List<ClassDeclarationSyntax>();
 var iTextStyleSymbol = compiliation.GetTypeByMetadataName("Microsoft.Maui.ITextStyle");

 foreach (var classDeclarationSyntax in collectedValues.CandidateClassesProvider)
 {
  var model = compiliation.GetSemanticModel(classDeclarationSyntax.SyntaxTree);
  var classSymbol = model.GetDeclaredSymbol(classDeclarationSyntax);

  if (classSymbol?.AllInterfaces.Any(i => i.OriginalDefinition.Equals(iTextStyleSymbol, SymbolEqualityComparer.Default)) ?? false)
   textStyleClassList.Add(classDeclarationSyntax);
}
```

Reply:
Argh! I've literally been pulling my hair out  for the last 2 hours trying to figure this out to find inherited interfaces using `IIncrementalGenerator `.

You're right that `AllInterfaces` does the trick, but I've only found samples using `ISourceGenerator` that demonstrate `AllInterfaces`.

There's gotta be a way to do this...